src/java.base/share/classes/sun/security/ssl/ServerHello.java

```diff
@@ -468,8 +471,12 @@ private static KeyExchangeProperties chooseCipherSuite(
                 }
             }
 
-            throw shc.conContext.fatal(Alert.HANDSHAKE_FAILURE,
-                    "no cipher suites in common");
```

Comment:
As there are detailed negotiation debug log, I may just update this line from "no cipher suites in common" to "no cipher suites or key exchange algorithms in common"

Reply:
See my previous reply.